storage/rocksdb/ha_rocksdb.cc

```diff
@@ -5516,7 +5686,7 @@ class Rdb_snapshot_status : public Rdb_tx_list_walker {
 
     /* Calculate the duration the snapshot has existed */
     int64_t snapshot_timestamp = tx->m_snapshot_timestamp;
-    if (snapshot_timestamp != 0) {
+    if (snapshot_timestamp != 0 && tx->has_snapshot()) {
```

Comment:
Some of this information still seems useful even if there's no actual snapshot on the transaction. It might be useful to also indicate whether this transaction has an active snapshot too.
I guess some of this information is in rocksdb_trx, but dogpiles only has show engine status.

Reply:
> Some of this information still seems useful even if there's no actual snapshot on the transaction. 

Are there any constraints on the output should look like in the case of no snapshot, if tooling parses it?

Reply:
Discussed offline